scalding-core/src/main/scala/com/twitter/scalding/typed/TypedPipe.scala

```diff
@@ -937,9 +937,17 @@ class TypedPipeInst[T] private[scalding] (@transient inpipe: Pipe,
       // for us. So unwind until you hit the first filter, snapshot,
       // then apply the unwound functions
       case Some((tap, fields, Converter(conv))) =>
-        Execution.from(new Iterable[T] {
-          def iterator = m.openForRead(conf, tap).asScala.map(tup => conv(tup.selectEntry(fields)))
-        })
+        /*
+           * We create a Stream here so we are lazy with construction, but we don't read twice.
+           */
+        val tryStream = scala.util.Try(m.openForRead(conf, tap)
+          .asScala
+          .map(tup => conv(tup.selectEntry(fields)))
+          .toStream)
```

Comment:
Hm, so I can remember being pretty happy when I realized that this was getting read separately on each mapper (though certainly, it's *much* to easy right now to accidentally read it 1000s of times on each mapper). I think it's easy to have data here that's small enough to fit in memory on each mapper but bigger than you really want to serialize into the JobConf. The ideal world would be for it to automatically get put into the distributedcache, but not sure how hard that is to arrange. Failing that, though, my vote would be to ensure that it is only read once per mapper, but not serialized by the submitter.